File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/model/instructions/InstructionTemplate.java
package org.hyperledger.iroha.android.model.instructions;

import org.hyperledger.iroha.android.model.InstructionBox;

/**
 * Produces an {@link InstructionBox}. Implementations capture the typed fields for each instruction
 * variant before converting to the argument-map representation used by local helpers. Transaction
 * encoding still requires wire-framed instruction payloads.
 */
public interface InstructionTemplate extends InstructionBox.InstructionPayload {

  default InstructionBox toInstructionBox() {
    return InstructionBox.of(this);
  }
}

File: .taira-source-c80bf4d85a97/kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/core/model/instructions/InstructionTemplate.kt
package org.hyperledger.iroha.sdk.core.model.instructions

import org.hyperledger.iroha.sdk.core.model.InstructionBox
import org.hyperledger.iroha.sdk.core.model.InstructionPayload

/**
 * Produces an `InstructionBox`. Implementations capture the typed fields for each instruction
 * variant before converting to the argument-map representation used by local helpers. Transaction
 * encoding still requires wire-framed instruction payloads.
 */
interface InstructionTemplate : InstructionPayload {

    fun toInstructionBox(): InstructionBox = InstructionBox.of(this)
}
